### PR TITLE
[Customer Portal] Restrict case creation and access for suspended and contract-expired projects

### PR DESCRIPTION
**Pull request by Hesara2003**
## Purpose
Prevent support case creation for projects whose contract has ended or whose closure state is suspended across both backend and webapp layers.

## Goals
- Reject case creation API requests (`POST /cases`) with HTTP 403 Forbidden when the target project's contract has expired or the project is suspended.
- Protect webapp project route boundaries (`ProjectGuard`) by displaying the project suspension notice when a project's contract end date has elapsed or when `closureState` is suspended.
- Suppress the "Get Help" navigation header dropdown for suspended and contract-expired projects.

## Approach
- **Frontend (`apps/customer-portal/webapp`):**
  - Added `isProjectContractEnded` in `src/utils/permission.ts` to evaluate whether a project's `endDate` has passed (inclusive through end of the calendar day in UTC).
  - Added `isProjectSuspended` in `src/utils/permission.ts` to perform case-insensitive checks on `closureState === "suspended"` and evaluate `isProjectContractEnded`.
  - Updated `ProjectGuard.tsx` to render `ProjectSuspendedNoticePage` when `isProjectSuspended(data?.closureState, data?.endDate)` evaluates to `true`.
  - Updated `GetHelpDropdown.tsx` to hide the "Get Help" dropdown if the project is suspended or expired.
- **Backend v1 (`apps/customer-portal/backend`):**
  - Added `isProjectContractEnded` and `isProjectSuspendedOrExpired` in `utils.bal`.
  - In `service.bal` resource `post cases`, query `entity:getProject(userInfo.idToken, payload.projectId)` before invoking `entity:createCase()`.
  - Return `http:Forbidden` (403) with message `"Cannot create cases for a suspended or contract-expired project."` when the project is suspended or its contract has expired.
- **Backend v2 (`apps/customer-portal/backend-v2`):**
  - Added `GetProject` to the `entityCaseClient` interface.
  - In `CaseHandler.CreateCase`, validate `ProjectID` UUID format, fetch project details, and evaluate `isProjectSuspendedOrExpired`.
  - Return HTTP 403 Forbidden when suspended or expired.
  - Added unit test suite in `internal/handler/cases_create_test.go` covering active, suspended, expired, and invalid UUID scenarios.

## User stories
As a Customer Portal administrator, I want inactive and expired projects to be blocked from creating new support tickets so that operations are not opened on lapsed contracts.

## Release note
- Block case creation and display suspension notices for suspended and contract-expired projects in Customer Portal.

## Documentation
N/A — internal Customer Portal behavior, no external doc surface affected.

## Automation tests
- **Webapp Vitest:**
  - `src/utils/__tests__/permission.test.ts`: 24/24 tests passed (contract end date comparisons, casing, edge cases).
  - `src/layouts/__tests__/ProjectGuard.test.tsx`: 5/5 tests passed (active, suspended, expired projects).
  - `src/components/header/__tests__/GetHelpDropdown.test.tsx`: 3/3 tests passed.
- **Go Backend v2 Tests:**
  - `internal/handler/cases_create_test.go`: All 4 test cases passed (`TestCreateCase_SuspendedProject_Forbidden`, `TestCreateCase_ExpiredProject_Forbidden`, `TestCreateCase_ActiveProject_Success`, `TestCreateCase_InvalidProjectID_BadRequest`).
  - Full suite `go test ./...` passed.
- **Ballerina Backend v1 Tests:**
  - `bal test` / `bal build`: Source compiled `wso2/customer_portal:1.0.0` with 0 errors.

## Security checks
- Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
- Ran FindSecurityBugs plugin and verified report? N/A (frontend and Go/Ballerina services)
- Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Test environment
- Node.js 20+ / pnpm 9+ / Vitest
- Go 1.22+
- Ballerina 2201.12.10 (Swan Lake Update 12)

## Summary by CodeRabbit

* **New Features**
  * Case creation now validates project IDs and blocks requests for suspended or contract-expired projects.
  * Get Help is hidden for suspended, expired, or restricted projects.
  * Project access handling recognizes suspension across closure states and contract end dates.
  * Onboarding values recognize additional “not applicable” formats.

* **Bug Fixes**
  * Contract expiration is detected through the end of the specified date.
  * Invalid project dates and identifiers receive appropriate validation responses.
  * Case feedback can only be submitted after the case is closed.

### CODE REVIEW

File: apps/customer-portal/backend/utils.bal [hunk 1]
```diff
@@ -1484,3 +1485,35 @@ public isolated function mapDeployedProductMetricsUsageCounts(
...
+    }
+    string trimmed = endDate.trim();
+    if trimmed.length() < 10 {
+        return false;
+    }
+    string endDatePart = trimmed.substring(0, 10);
+    // End date is considered inclusive through the end of the specified day (UTC).
+    // The contract has ended if current UTC date is strictly greater than the end date.
+    string currentDatePart = time:utcToString(time:utcNow()).substring(0, 10);
+    return currentDatePart > endDatePart;
```
[the remark this answers is not in the document]
Comment (Hesara2003) [reply apps/customer-portal/backend/utils.bal:1519]: Addressed in commits 07ccd2f7b (webapp) and 983edb617 (backend):
- In `apps/customer-portal/webapp/src/utils/permission.ts`, validated that `Date.UTC` values match the original year, month, and day to prevent rollover on calendar-invalid dates (e.g. `2026-02-30` or non-leap-year `2026-02-29`), and added corresponding test cases in `permission.test.ts`.
- In `apps/customer-portal/backend/utils.bal`, used Ballerina's `time:dateValidate({year, month, day})` before comparing calendar dates.

**Comment by Hesara2003** [comment]
@coderabbitai review

**Review by Rashmika998 — APPROVED** [review]
